client/my-sites/comments/comment-list/index.jsx

```diff
@@ -198,32 +224,54 @@ export class CommentList extends Component {
 			return;
 		}
 
-		const options = {
+		const noticeOptions = {
 			button: translate( 'Undo' ),
 			duration: 5000,
 			id: `comment-notice-${ commentId }`,
 			isPersistent: true,
-			onClick: () => this.setCommentStatus( commentId, postId, previousStatus, { showNotice: false } ),
+			onClick: () => this.setCommentStatus( commentId, postId, previousStatus, {
+				isUndo: true,
+				persist: options.persist,
+				showNotice: false,
+			} ),
 		};
 
-		this.props.createNotice( type, message, options );
+		this.props.createNotice( type, message, noticeOptions );
 	}
 
 	toggleBulkEdit = () => this.setState( { isBulkEdit: ! this.state.isBulkEdit } );
 
 	toggleCommentLike = ( commentId, postId ) => {
 		// TODO: Replace with Redux getComment()
-		const comment = find( this.props.comments, [ 'ID', commentId ] );
-
-		if ( 'unapproved' === comment.status ) {
-			this.props.removeNotice( `comment-notice-${ commentId }` );
-			this.setCommentStatus( commentId, postId, 'approved' );
-		}
+		const comment = this.getComment( commentId );
+		const isPersisted = !! find( this.state.persistedComments, [ 'ID', commentId ] );
 
 		if ( comment.i_like ) {
 			this.props.unlikeComment( commentId, postId );
+
+			if ( isPersisted ) {
+				this.updatePersistedComments( {
+					...comment,
+					i_like: false,
+				} );
+			}
 		} else {
 			this.props.likeComment( commentId, postId );
+
+			if ( 'unapproved' === comment.status ) {
+				this.props.removeNotice( `comment-notice-${ commentId }` );
+				this.setCommentStatus( commentId, postId, 'approved' );
+				this.updatePersistedComments( {
+					...comment,
+					i_like: true,
+					status: 'approved',
+				} );
+			} else if ( isPersisted ) {
+				this.updatePersistedComments( {
+					...comment,
+					i_like: true,
+				} );
+			}
 		}
 	}
```

Comment:
great googly moogly @copons I found the collection of `}`s I have been missing!

do you think we could explore un-nested early-abort measures instead of deeply-nesting here? check out the complexity we're creating when we branch for each condition…